File: frontend/src/lib/institution.ts
import { apiRequest } from "@/lib/api";
import { SharingCategory } from "@/lib/emotional";
import { UserRole } from "@/types/auth";

export type InstitutionCategoryCount = {
  category: SharingCategory;
  count: number;
};

export type InstitutionMoodCount = {
  mood: string;
  count: number;
};

export type InstitutionSharedUserSummary = {
  user_id: string;
  full_name: string;
  email: string;
  categories: SharingCategory[];
  summary_only: boolean;
  latest_mood: string | null;
  average_intensity: number | null;
  journal_entries_visible: number;
  shared_at: string;
};

export type InstitutionDashboard = {
  institution_role: UserRole;
  participant_count: number;
  category_breakdown: InstitutionCategoryCount[];
  mood_breakdown: InstitutionMoodCount[];
  average_intensity: number | null;
  average_anxiety: number | null;
  average_stress: number | null;
  risk_flags: number;
  summary: string;
  privacy_note: string;
  shared_users: InstitutionSharedUserSummary[];
  generated_at: string;
};

export async function getInstitutionDashboard(): Promise<InstitutionDashboard> {
  return apiRequest<InstitutionDashboard>("/institution/dashboard");
}

File: backend/app/schemas/institution.py
from datetime import datetime

from pydantic import BaseModel, EmailStr

from app.models.emotional import SharingCategory
from app.models.user import UserRole


class InstitutionCategoryCount(BaseModel):
    category: SharingCategory
    count: int


class InstitutionMoodCount(BaseModel):
    mood: str
    count: int


class InstitutionSharedUserSummary(BaseModel):
    user_id: str
    full_name: str
    email: EmailStr
    categories: list[SharingCategory]
    summary_only: bool
    latest_mood: str | None
    average_intensity: float | None
    journal_entries_visible: int
    shared_at: datetime


class InstitutionDashboardResponse(BaseModel):
    institution_role: UserRole
    participant_count: int
    category_breakdown: list[InstitutionCategoryCount]
    mood_breakdown: list[InstitutionMoodCount]
    average_intensity: float | None
    average_anxiety: float | None
    average_stress: float | None
    risk_flags: int
    summary: str
    privacy_note: str
    shared_users: list[InstitutionSharedUserSummary]
    generated_at: datetime
